File: Selenium/Activities/Activity14.java
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import java.util.List;
public class Activity14 {
    public static void main(String[] args) {
        WebDriverManager.chromedriver().setup();
        WebDriver driver = new ChromeDriver();
        driver.get("https://v1.training-support.net/selenium/tables");
        System.out.println(driver.getTitle());
        List<WebElement> rows=driver.findElements(By.xpath("//table[@id='sortableTable']/tbody/tr"));
        System.out.println("number of rows : "+rows.size());
        List<WebElement> cols=driver.findElements(By.xpath("//table[@id='sortableTable']/tbody/tr[1]/td"));
        System.out.println("number of columns  : "+cols.size());
        WebElement secondColValue=driver.findElement(By.xpath("//table[@id='sortableTable']/tbody/tr[2]/td[2]"));
        System.out.println("second row second column values : "+secondColValue.getText());
        driver.findElement(By.xpath("//table[@id='sortableTable']//thead//tr//th[1]")).click();
        System.out.println("second row second column values : "+secondColValue.getText());
        List<WebElement> footerValues=driver.findElements(By.xpath("//table[@id='sortableTable']/tfoot/tr/th"));
        for(WebElement cells:footerValues){
            System.out.println("Footer values : "+cells.getText());
        }
        driver.close();
    }
}

File: Selenium/Activities/Activity14.py
from selenium import webdriver
from selenium.webdriver.chrome.service import Service as chromeService
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver import ActionChains
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
service=chromeService(ChromeDriverManager().install())
with webdriver.Chrome(service=service) as driver:
    actions = webdriver.ActionChains(driver)
    driver.get("https://v1.training-support.net/selenium/tables")
    print(driver.title)
    rows = driver.find_elements(By.XPATH, "//table[@id='sortableTable']/tbody/tr")
    print("Number of rows: ", len(rows))
    columns = driver.find_elements(By.XPATH, "//table[@id='sortableTable']/tbody/tr[1]/td")
    print("Number of rows: ", len(columns))
    secondColValue=driver.find_element(By.XPATH, "//table[@id='sortableTable']/tbody/tr[2]/td[2]")
    print("second row second column values : ",secondColValue.text)
    driver.find_element(By.XPATH,"//table[@id='sortableTable']//thead//tr//th[1]").click();
    print("second row second column values :",secondColValue.text)
    footerValues=driver.find_elements(By.XPATH,"//table[@id='sortableTable']/tfoot/tr/th");
    for row in footerValues:
        print("footer value : ",row.text)
    driver.close()

